sphinx/ext/autodoc/__init__.py

```diff
@@ -1809,12 +1809,14 @@ def import_object(self, raiseerror: bool = False) -> bool:
         except ImportError as exc:
             # annotation only instance variable (PEP-526)
             try:
-                self.parent = importlib.import_module(self.modname)
-                annotations = get_type_hints(self.parent, None,
-                                             self.config.autodoc_type_aliases)
-                if self.objpath[-1] in annotations:
-                    self.object = UNINITIALIZED_ATTR
-                    return True
+                with mock(self.config.autodoc_mock_imports):
+                    parent = importlib.import_module(self.modname)
```

Comment:
I noticed we have to use `sphinx.ext.autodoc.importer:import_module()` instead of importlib's.